liota/transports/mqtt.py

```diff
@@ -48,25 +49,78 @@ def on_message(self, client, userdata, msg):
     def on_publish(self, client, userdata, mid):
         log.debug("mid: {0}".format(str(mid)))
 
-    def __init__(self, url, port):
+    def on_connect(self, client, userdata, flags, rc):
+        log.debug("Connected with result code " + str(rc))
+
+    def __init__(self, gw_identity, TLSConf, url, port, tls_support=True):
+        self.gw_identity = gw_identity
+        self.TLSConf = TLSConf
         self.url = url
         self.port = port
         self.client = paho.Client()
         self.client.on_message = self.on_message
         self.client.on_publish = self.on_publish
         self.client.on_subscribe = self.on_subscribe
+        self.tls_support = tls_support
         self.connect_soc()
         TransportLayer.__init__(self)
 
     def connect_soc(self):
-        self.client.connect_soc(host=self.url, port=self.port, keepalive=60)
-        log.info("Connection Successful")
+        # Set up TLS support
+        if self.tls_support:
+
+            # Validate CA certificate path
+            if self.gw_identity.cacert:
+                if not(os.path.exists(self.gw_identity.cacert)):
+                    raise ValueError("Error : Wrong CA certificate path.")
+            else:
+                raise ValueError("Error : CA certificate path is missing")
+
+            # Validate client certificate path
+            if self.gw_identity.certfile:
+                if os.path.exists(self.gw_identity.certfile):
+                    client_cert_available = True
+                else:
+                    raise ValueError("Error : Wrong client certificate path.")
+            else:
+                client_cert_available = False
+
+            # Validate client key file path
+            if self.gw_identity.keyfile:
+                if os.path.exists(self.gw_identity.keyfile):
+                    client_key_available = True
+                else:
+                    raise ValueError("Error : Wrong client key path.")
+            else:
+                client_key_available = False
+
+            # Multiple conditions for certificate validations
+                # 1. Both Client certificate and key file should be present
+                # 2. If both are not there proceed without client certificate and key
+                # 3. If one of them is not there throw an error to notify one of them is missing
+            if client_cert_available != client_key_available:
```

Comment:
1) Boolean operator "not" preferable over "!="
2) Also, think of splitting the first case into two; a) "Client cert file found. But key not found." b) "Client key file found. But cert file not found."